tape/src/test/java/com/squareup/tape/QueueFileTest.java

```diff
@@ -188,6 +189,37 @@
     assertThat(queue.peek()).isEqualTo(secondStuff);
   }
 
+  @Test public void removeNonPositiveNumberOfElementsComplains() throws IOException {
+    QueueFile queue = new QueueFile(file);
+    queue.add(values[127]);
+
+    try {
+      queue.remove(-1);
+      fail("Should have complained about removing negative number of elements.");
+    } catch (IllegalArgumentException ex) {
+      assertThat(ex) //
+          .hasMessage("Cannot remove a non-positive (-1) number of elements.");
+    }
+
+    try {
+      queue.remove(0);
```

Comment:
I'd like to no-op this case because you might be doing math to calculate the removal of tasks and it saves you from checking against 0.